updater.js

```diff
@@ -48,12 +56,57 @@ class AlAzureUpdater {
             return new ApplicationTokenCredentials(this._clientId, this._domain, this._clientSecret);
         }
     }
-    
+
     syncWebApp(callback) {
         const credentials = this._getAzureCredentials();
         const webSiteClient = new WebSiteManagementClient(credentials, this._subscriptionId);
         return webSiteClient.webApps.syncRepository(this._resourceGroup, this._webAppName, callback);
     }
+
+    setEnvConfigChanges(envObject, callback) {
+        var updateEnv = envObject;
+        m_util.updateAppSettings(updateEnv, this.azureWebsiteClientObject, function (settingsError) {
+            if (settingsError) {
+                return callback(settingsError);
+            } else {
+                return callback(null);
+            }
+        });
+    }
+    
+    readEnvFile(callback){
+        fs.readFile(process.cwd() + "/"+ process.env.AZURE_FUN_UPDATE_CONFIG_NAME,'utf-8',function (err, data) {
+          if (err) {
+             return callback(err);
+            } else{
+              const envData = JSON.parse(data);
```

Comment:
please wrap `JSON.parse` into `try..catch` and return `callback(exception)`

Reply:
resolved